File: HW/HomeWork/HomeWork/TextContainer.cpp
#pragma once
#include <vector>
#include <string>
#include "TextContainer.h"

	TextContainer::TextContainer(const std::vector<std::string>& inputLines) {
		lines = inputLines;
	}

	int TextContainer::operator[](int index) {
		if (index >= 0 && index < lines.size()) {
			return lines[index].size();
		}
		else {
			return -1;
		}
	}

	int TextContainer::getNumber() {
		int count = 0;
		char vowels[5] = { 'a', 'i', 'o', 'e', 'u' };

		for (std::string& line : lines) {
			for (char ch : line) {
				for (int i = 0;i < 5;i++) {
					if (ch == vowels[i]) {
						count++;
					}
				}
			}
		}

		return count;
	}

File: HW/HomeWork_/HomeWork_/TextContainer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HomeWork_
{
    internal class TextContainer
    {
        private List<string> text;
        public TextContainer(List<string> lines)
        {
            text = lines;
        }

        public int this[int index]
        {
            get
            {
                if (index < 0 || index >= text.Count)
                    throw new IndexOutOfRangeException("Невірний індекс");
                return text[index].Length;
            }
        }

        public int getNumberVowels()
        {
            int count = 0;
            char[] vowels = { 'a', 'i', 'e', 'o', 'u' };

            foreach (string line in text)
            {
                foreach (char ch in line)
                {
                    for (int i = 0; i < 5; i++)
                    {
                        if (ch == vowels[i])
                        {
                            count++;
                        }
                    }
                }
            }

            return count;

        }
    }
}
